staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -2862,6 +2862,31 @@ type PodDNSConfigOption struct {
 	Value *string `json:"value,omitempty" protobuf:"bytes,2,opt,name=value"`
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// EphemeralContainer describes a container to attach to a running pod for troubleshooting.
+// This field is alpha-level and is only honored by servers that enable the DebugContainers feature.
+type EphemeralContainer struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Spec describes the Ephemeral Container to be created.
+	Spec *Container `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+
+	// Most recently observed status of the container.
+	// This data may not be up to date.
+	// Populated by the system.
+	// Read-only.
+	// +optional
+	Status *ContainerStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
```

Comment:
You are only allowed to have one ephemeral container at a time, I take it? Otherwise I'm not sure how this can work. Might be good to state how that works in the comments.

Reply:
OK, I understand now, I didn't realize it wasn't a subresource.